### PR TITLE
Add nil check operator in token checking at token introspection

### PR DESCRIPTION
**Pull request by linhdangduy**
### Summary
#1242 fix

### CODE REVIEW

**Comment by linhdangduy** [comment]
@nbulaj 
I have just add NEWS.md and fix typo of one test. Is that ok?
Should I write test for this change?

**Comment by nbulaj** [comment]
@linhdangduy it would be great to have a test for such case, so yes if you can

**Comment by linhdangduy** [comment]
@nbulaj I added the test for this case. Is that clear?

File: lib/doorkeeper/oauth/token_introspection.rb [hunk 1]
```diff
@@ -163,7 +163,7 @@ def valid_token?
 
       # RFC7662 Section 2.1
       def authorized_token_matches_introspected?
-        authorized_token.token == @token.token
+        authorized_token.token == @token&.token
```
Comment (manhdaovan) [inline lib/doorkeeper/oauth/token_introspection.rb]: Cause `&.` operator is support from ruby ver. >= 2.3, so is this ok if ruby version is < 2.3?

**Comment by nbulaj** [comment]
@linhdangduy yep, thanks!

**Pull request merged**